tests/cupyx_tests/distributed_tests/test_comm.py

```diff
@@ -0,0 +1,286 @@
+import time
+import unittest
+import pytest
+
+import cupy
+from cupy import cuda
+from cupy.cuda import nccl
+from cupy import testing
+
+from cupyx.distributed import init_process_group
+from cupyx.distributed._nccl_comm import NCCLBackend
+from cupyx.distributed._store import ExceptionAwareProcess
+
+nccl_available = nccl.available
+
+
+N_WORKERS = 2
+
+
+@pytest.mark.skipif(not nccl_available, reason='nccl is not installed')
+class TestNCCLBackend:
+    def _launch_workers(self, n_workers, func, args=()):
+        processes = []
+        # TODO catch exceptions
+        for rank in range(n_workers):
+            p = ExceptionAwareProcess(
+                target=func,
+                args=(rank, n_workers) + args)
+            p.start()
+            processes.append(p)
+
+        for p in processes:
+            p.join()
+
+    @testing.for_all_dtypes(no_bool=True)
+    def test_broadcast(self, dtype):
+        if dtype in (cupy.int16, cupy.uint16):
+            return  # nccl does not support int16
+
+        def run_broadcast(rank, n_workers, root, dtype):
+            dev = cuda.Device(rank)
+            dev.use()
+            comm = NCCLBackend(n_workers, rank)
+            expected = cupy.arange(2 * 3 * 4, dtype=dtype).reshape((2, 3, 4))
+            if rank == root:
+                in_array = expected
+            else:
+                in_array = cupy.zeros((2, 3, 4), dtype=dtype)
+
+            comm.broadcast(in_array, root)
+            testing.assert_allclose(in_array, expected)
+
+        self._launch_workers(N_WORKERS, run_broadcast, (0, dtype))
+        self._launch_workers(N_WORKERS, run_broadcast, (1, dtype))
```

Comment:
This looks strange... Don't you already have a loop in `_launch_workers` to generate rank 0 & 1? Why do we need to make 2 calls here?

Reply:
it launches the broadcast two times, with a different root process.
Could have parametrized it but was not in the mood I guess.

Reply:
Ah OK, thanks for the explanation!